netbox/dcim/tests/test_cablepaths.py

```diff
@@ -2343,3 +2344,41 @@ def test_401_exclude_midspan_devices(self):
             is_active=True
         )
         self.assertEqual(CablePath.objects.count(), 0)
+
+    def test_detect_infinite_loop(self):
```

Comment:
Is this test suppose to fail or succeed?  As it stands I don't see a loop in the topology unuless I am missing something.

Reply:
If there's an infinite loop then the test will just keep executing. I'm not sure of an elegant way to test the infinite looping condition, but as long as it executes without the whole test process hanging then the test passes.

I've tried a few different approaches for making it assert if the execution takes too long or loops too many times, mostly involving timeouts and/or threading, but I don't know if it's worth going to that extreme. What would you suggest?

In any case if you comment out lines 600-602 in cables.py, the test will hang indefinitely.

Reply:
IMO, for testing purposes we may want to add a parameter that sets a maximum path length (might not be bad anyways to set it to an unreasonable number) and still create a loop.

IDK if it is worth it though TBH.